PolyFill.cs

```diff
@@ -0,0 +1,59 @@
+﻿//// PolyFill.cs - Contains the PolyFill class which is mainly used to 
+//// fill a given polygon
+//// ---------------------------------------------------------------------------------------
+namespace GrayBMP;
+
+#region class PolyFill ---------------------------------------------------------
+/// <summary>Class to read a given polygon and to fill it </summary>
+class PolyFill {
+   public void AddLine (int x0, int y0, int x1, int y1) {
+      if (y0 == y1) return; // Don't add horizontal lines
+      // We will try to keep Y0 as the smaller one always
+      if (y0 > y1) { (y0, y1) = (y1, y0); (x0, x1) = (x1, x0); }
+      mLines.Add (new Line (x0, y0, x1, y1));
+   }
+
+   public void Fill (GrayBMP bmp, int color) {
+      mLines.Sort ();
+      // Current active lines along with their slopes for the given scan and the intersections with the current scanline
+      var (aLines, ixs) = (new List<(Line Ln, double Slope)> (), new List<int> ());
+      for (int y = 0; y < bmp.Height; y++) {
+         ixs.Clear ();
+         var ys = y + 0.5;
+         // Remove lines that are out of the scope of this scan
+         for (int i = aLines.Count - 1; i >= 0; i--)
+            if (aLines[i].Ln.Y1 < ys) aLines.RemoveAt (i);
+         // Add lines that come into the scope of this scan
+         while (mLines.Any () && mLines[0].Y0 < ys) {
+            var ln = mLines[0];
+            aLines.Add ((ln, (double)(ln.X1 - ln.X0) / (ln.Y1 - ln.Y0)));
+            mLines.RemoveAt (0);

```

Comment:
This is very inefficient. Removing the 0th element in a list will need to move all the remaining elements down by 1 and will take O(n) time each time. Instead, why don't you have a variable that tracks how many elements from the list you have 'used' already, and just keep incrementing that counter?
Alternatively, reverse-sort the list and remove elements from the end - that is very fast, since no elements will need to be moved.

Reply:
Done. Thanks for the time